src/components/scripture/labels.ts

```diff
@@ -88,7 +91,16 @@ export const labelOfScriptureRanges = (refs: readonly ScriptureRange[]) => {
   const hasSame = (key: keyof ScriptureReference) =>
     uniq(refs.flatMap((ref) => [ref.start[key], ref.end[key]])).length === 1;
   const same = hasSame('book') ? 'book' : undefined;
-  const labels = refs.map((ref) => labelOfScriptureRange(ref, same)).join(', ');
+  const labels = refs.map((ref) => labelOfScriptureRange(ref, same));
   const prefix = same === 'book' ? `${refs[0].start.book} ` : '';
-  return `${prefix}${labels}`;
+  const collapseLabels = (labels: string[]) => {
+    if (collapseAfter && labels.length > collapseAfter) {
+      const collapsed = labels.slice(0, collapseAfter);
+      collapsed.push(`and ${labels.length - collapseAfter} other portions`);
+      return collapsed.join(', ');
+    }
+    return labels.join(', ');
```

Comment:
I found out about the Intl.ListFormat API recently. It could help format this string.
```tsx
const labelOutput = new Intl.ListFormat().format(labels)
```
Would still need to throw in the last segment.
```ts
[
   ...labels,
   ...(labels.length > collapseAfter ? [`${labels.length - collapseAfter} other portions`] : []),
]
```

Reply:
Just noting that I made a const to keep the original length of the refs because we lose that on the split